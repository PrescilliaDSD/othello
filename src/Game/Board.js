import React, { useState, useLayoutEffect } from 'react'
import PropTypes from 'prop-types'

import Chip from './Chip'

const Board = ({ currentPlayer }) => {
  const [board, setBoard] = useState([
    ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
    ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
    ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
    ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
    ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
    ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
    ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
    ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
  ])

  const addChip = async (event) => {
    const coordinates = await event.target.id.split('')
    const newBoard = await board.map((row, i) => {
      return row.map((column, j) => {
        if (i === coordinates[0] - 1 && j === coordinates[1] - 1) {
          return 'black'
        }
        return column
      })
    })
    await parseTopToBottom(newBoard)
  }

  const parseTopToBottom = (boardUpdated) => {
    const boardParsed = boardUpdated.map((row, i) => {
      if (i <= 5) {
        return row.map((column, j) => {
          if (column === 'white') {
            return 'white'
          } else if (column === 'black') {
            return 'black'
          } else if (boardUpdated[i + 1][j] === 'no' || boardUpdated[i + 1][j] === 'black' || boardUpdated[i + 1][j] === 'available') {
            return column
          } else if (currentPlayer === 'black' && boardUpdated[i + 1][j] === 'white' && boardUpdated[i + 2][j] === 'black') {
            return 'available'
          }

        })
      } else {
        return row
      }
    })
    setBoard(boardParsed)
  }

  return (
    <>
      <article className="board">
        <ul className="board__row">
          <li className="row-number"></li>
          <li className="row-number">1</li>
          <li className="row-number">2</li>
          <li className="row-number">3</li>
          <li className="row-number">4</li>
          <li className="row-number">5</li>
          <li className="row-number">6</li>
          <li className="row-number">7</li>
          <li className="row-number">8</li>
        </ul>
        {
          board.map((row, i) => (
            <ul key={`row${i}`} className="board__row">
              <li className="column-number">{ i + 1 }</li>
              {
                row.map((column, j) => {
                  if (column === 'available') {
                    return (
                      <li onClick={addChip} className="board__square clickable" key={`row${i}column${j}`} id={`${i + 1}${j + 1}`}>
                        <Chip color={column} />
                      </li>
                    )
                  }
                  return (
                    <li className="board__square" key={`row${i}column${j}`} id={`${i + 1}${j + 1}`}>
                      <Chip color={column} />
                    </li>
                  )
                })
              }
            </ul>
          ))
        }
      </article>
    </>
  )
}

Board.propTypes = {
  currentPlayer: PropTypes.string.isRequired
}

export default Board
