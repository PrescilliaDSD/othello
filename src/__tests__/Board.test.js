import React from 'react'
import { StaticRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Board from '../components/Game/Board'

Enzyme.configure({ adapter: new Adapter() })

describe('Board component initialization', () => {
  it('should render correctly', () => {
    const component = renderer
      .create(<StaticRouter params="123"><Board opponent='' setOpponent={() => {}} setBoard={() => {}} setLastPlayerPass={() => {}} setWhiteScore={() => {}} setBlackScore={() => {}} endGame={() => {}} player1='' player2='' status='' whiteScore={0} blackScore={0} lastPlayerPass='' board={[
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
      ]}
      currentPlayer={'black'} setCurrentPlayer={() => {}} /></StaticRouter>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render a black chip in square 45', () => {
    const component = renderer
      .create(<StaticRouter params="123"><Board opponent='' setOpponent={() => {}} setBoard={() => {}} setLastPlayerPass={() => {}} setWhiteScore={() => {}} setBlackScore={() => {}} endGame={() => {}} player1='' player2='' status='' whiteScore={0} blackScore={0} lastPlayerPass='' board={[
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
      ]}
      currentPlayer={'black'} setCurrentPlayer={() => {}} /></StaticRouter>)
    const tree = component.toJSON()
    expect(tree.children[4].children[5].children[0].props.className).toBe('black-chip chip')
  })

  it('should render a black chip in square 54', () => {
    const component = renderer
      .create(<StaticRouter params="123"><Board opponent='' setOpponent={() => {}} setBoard={() => {}} setLastPlayerPass={() => {}} setWhiteScore={() => {}} setBlackScore={() => {}} endGame={() => {}} player1='' player2='' status='' whiteScore={0} blackScore={0} lastPlayerPass='' board={[
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
      ]}
      currentPlayer={'black'} setCurrentPlayer={() => {}} /></StaticRouter>)
    const tree = component.toJSON()
    expect(tree.children[5].children[4].children[0].props.className).toBe('black-chip chip')
  })

  it('should render a white chip in square 44', () => {
    const component = renderer
      .create(<StaticRouter params="123"><Board opponent='' setOpponent={() => {}} setBoard={() => {}} setLastPlayerPass={() => {}} setWhiteScore={() => {}} setBlackScore={() => {}} endGame={() => {}} player1='' player2='' status='' whiteScore={0} blackScore={0} lastPlayerPass='' board={[
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
      ]}
      currentPlayer={'black'} setCurrentPlayer={() => {}} /></StaticRouter>)
    const tree = component.toJSON()
    expect(tree.children[4].children[4].children[0].props.className).toBe('white-chip chip')
  })

  it('should render a white chip in square 55', () => {
    const component = renderer
      .create(<StaticRouter params="123"><Board opponent='' setOpponent={() => {}} setBoard={() => {}} setLastPlayerPass={() => {}} setWhiteScore={() => {}} setBlackScore={() => {}} endGame={() => {}} player1='' player2='' status='' whiteScore={0} blackScore={0} lastPlayerPass='' board={[
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
      ]}
      currentPlayer={'black'} setCurrentPlayer={() => {}} /></StaticRouter>)
    const tree = component.toJSON()
    expect(tree.children[5].children[5].children[0].props.className).toBe('white-chip chip')
  })

  it('should render an available chip in square 34', () => {
    const component = renderer
      .create(<StaticRouter params="123"><Board opponent='' setOpponent={() => {}} setBoard={() => {}} setLastPlayerPass={() => {}} setWhiteScore={() => {}} setBlackScore={() => {}} endGame={() => {}} player1='' player2='' status='' whiteScore={0} blackScore={0} lastPlayerPass='' board={[
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
      ]}
      currentPlayer={'black'} setCurrentPlayer={() => {}} /></StaticRouter>)
    const tree = component.toJSON()
    expect(tree.children[3].children[4].children[0].props.className).toBe('available-chip chip')
  })

  it('should render an available chip in square 43', () => {
    const component = renderer
      .create(<StaticRouter params="123"><Board opponent='' setOpponent={() => {}} setBoard={() => {}} setLastPlayerPass={() => {}} setWhiteScore={() => {}} setBlackScore={() => {}} endGame={() => {}} player1='' player2='' status='' whiteScore={0} blackScore={0} lastPlayerPass='' board={[
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
      ]}
      currentPlayer={'black'} setCurrentPlayer={() => {}} /></StaticRouter>)
    const tree = component.toJSON()
    expect(tree.children[4].children[3].children[0].props.className).toBe('available-chip chip')
  })

  it('should render an available chip in square 56', () => {
    const component = renderer
      .create(<StaticRouter params="123"><Board opponent='' setOpponent={() => {}} setBoard={() => {}} setLastPlayerPass={() => {}} setWhiteScore={() => {}} setBlackScore={() => {}} endGame={() => {}} player1='' player2='' status='' whiteScore={0} blackScore={0} lastPlayerPass='' board={[
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
      ]}
      currentPlayer={'black'} setCurrentPlayer={() => {}} /></StaticRouter>)
    const tree = component.toJSON()
    expect(tree.children[5].children[6].children[0].props.className).toBe('available-chip chip')
  })

  it('should render an available chip in square 65', () => {
    const component = renderer
      .create(<StaticRouter params="123"><Board opponent='' setOpponent={() => {}} setBoard={() => {}} setLastPlayerPass={() => {}} setWhiteScore={() => {}} setBlackScore={() => {}} endGame={() => {}} player1='' player2='' status='' whiteScore={0} blackScore={0} lastPlayerPass='' board={[
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'available', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'available', 'white', 'black', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'black', 'white', 'available', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'available', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no'],
        ['no', 'no', 'no', 'no', 'no', 'no', 'no', 'no']
      ]}
      currentPlayer={'black'} setCurrentPlayer={() => {}} /></StaticRouter>)
    const tree = component.toJSON()
    expect(tree.children[6].children[5].children[0].props.className).toBe('available-chip chip')
  })
})
