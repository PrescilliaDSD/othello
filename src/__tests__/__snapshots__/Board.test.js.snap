// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component initialization should render correctly 1`] = `
<article
  className="board"
>
  <ul
    className="board__row"
  >
    <li
      className="row-number"
    />
    <li
      className="row-number"
    >
      1
    </li>
    <li
      className="row-number"
    >
      2
    </li>
    <li
      className="row-number"
    >
      3
    </li>
    <li
      className="row-number"
    >
      4
    </li>
    <li
      className="row-number"
    >
      5
    </li>
    <li
      className="row-number"
    >
      6
    </li>
    <li
      className="row-number"
    >
      7
    </li>
    <li
      className="row-number"
    >
      8
    </li>
  </ul>
  <ul
    className="board__row"
  >
    <li
      className="column-number"
    >
      1
    </li>
    <li
      className="board__square"
      data-testid="11"
      id="11"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="12"
      id="12"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="13"
      id="13"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="14"
      id="14"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="15"
      id="15"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="16"
      id="16"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="17"
      id="17"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="18"
      id="18"
    >
      <div
        className="no-chip chip"
      />
    </li>
  </ul>
  <ul
    className="board__row"
  >
    <li
      className="column-number"
    >
      2
    </li>
    <li
      className="board__square"
      data-testid="21"
      id="21"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="22"
      id="22"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="23"
      id="23"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="24"
      id="24"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="25"
      id="25"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="26"
      id="26"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="27"
      id="27"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="28"
      id="28"
    >
      <div
        className="no-chip chip"
      />
    </li>
  </ul>
  <ul
    className="board__row"
  >
    <li
      className="column-number"
    >
      3
    </li>
    <li
      className="board__square"
      data-testid="31"
      id="31"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="32"
      id="32"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="33"
      id="33"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square clickable"
      data-testid="34"
      id="34"
      onClick={[Function]}
    >
      <div
        className="available-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="35"
      id="35"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="36"
      id="36"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="37"
      id="37"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="38"
      id="38"
    >
      <div
        className="no-chip chip"
      />
    </li>
  </ul>
  <ul
    className="board__row"
  >
    <li
      className="column-number"
    >
      4
    </li>
    <li
      className="board__square"
      data-testid="41"
      id="41"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="42"
      id="42"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square clickable"
      data-testid="43"
      id="43"
      onClick={[Function]}
    >
      <div
        className="available-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="44"
      id="44"
    >
      <div
        className="white-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="45"
      id="45"
    >
      <div
        className="black-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="46"
      id="46"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="47"
      id="47"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="48"
      id="48"
    >
      <div
        className="no-chip chip"
      />
    </li>
  </ul>
  <ul
    className="board__row"
  >
    <li
      className="column-number"
    >
      5
    </li>
    <li
      className="board__square"
      data-testid="51"
      id="51"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="52"
      id="52"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="53"
      id="53"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="54"
      id="54"
    >
      <div
        className="black-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="55"
      id="55"
    >
      <div
        className="white-chip chip"
      />
    </li>
    <li
      className="board__square clickable"
      data-testid="56"
      id="56"
      onClick={[Function]}
    >
      <div
        className="available-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="57"
      id="57"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="58"
      id="58"
    >
      <div
        className="no-chip chip"
      />
    </li>
  </ul>
  <ul
    className="board__row"
  >
    <li
      className="column-number"
    >
      6
    </li>
    <li
      className="board__square"
      data-testid="61"
      id="61"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="62"
      id="62"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="63"
      id="63"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="64"
      id="64"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square clickable"
      data-testid="65"
      id="65"
      onClick={[Function]}
    >
      <div
        className="available-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="66"
      id="66"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="67"
      id="67"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="68"
      id="68"
    >
      <div
        className="no-chip chip"
      />
    </li>
  </ul>
  <ul
    className="board__row"
  >
    <li
      className="column-number"
    >
      7
    </li>
    <li
      className="board__square"
      data-testid="71"
      id="71"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="72"
      id="72"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="73"
      id="73"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="74"
      id="74"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="75"
      id="75"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="76"
      id="76"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="77"
      id="77"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="78"
      id="78"
    >
      <div
        className="no-chip chip"
      />
    </li>
  </ul>
  <ul
    className="board__row"
  >
    <li
      className="column-number"
    >
      8
    </li>
    <li
      className="board__square"
      data-testid="81"
      id="81"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="82"
      id="82"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="83"
      id="83"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="84"
      id="84"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="85"
      id="85"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="86"
      id="86"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="87"
      id="87"
    >
      <div
        className="no-chip chip"
      />
    </li>
    <li
      className="board__square"
      data-testid="88"
      id="88"
    >
      <div
        className="no-chip chip"
      />
    </li>
  </ul>
</article>
`;
